streamer/input_configuration.py

```diff
@@ -146,6 +146,9 @@ class Input(configuration.Base):
   Otherwise, it will default to 'und' (undetermined).
   """
 
+  channels = configuration.Field(int).cast()
+  """The number of audio channels to encode."""
```

Comment:
It's more like "The number of channels in an audio input", right?  It's less about encoding, and more about the structure of the input.  Do I understand that right?

Reply:
No, i feel a bit confused because of this too.
This is actually an output feature, the number of channels to encode in the final output file. sticking it with the each input(instead of the pipeline config) will allow us to give different channel counts for each input in a multi-period list.

Reply:
Another less confusing approach could be, to deal with it like resolutions, we may specify the channel counts we want as an output to be encoded with, and it's a list of channel counts rather than a single channel count and each input will be encoded with different channel counts up till its own channel count just like video resolutions.

Reply:
That would make sense to me.  I like having it in the input, though, with auto-detection, just like resolution and frame rate for video inputs.  If auto-detect fails, or if the input is coming from a pipe, rather than a file, we would need to be able to specify it in the input config.

So let's keep what you have (but fix the comment here on the field meaning), and then let's add a list of output channel counts in the pipeline config, just like we do for resolutions.